lingua_franca/lang/common_data_pt.py

```diff
@@ -116,3 +148,174 @@
+    (1e42, "tredecilião"),
+    (1e45, "quattuordecilião"),
+    (1e48, "quinquadecilião"),
+    (1e51, "sedecilião"),
+    (1e54, "septendecilião"),
+    (1e57, "octodecilião"),
+    (1e60, "novendecilião"),
+    (1e63, "vigintilião"),
+    (1e66, "unvigintilião"),
+    (1e69, "uuovigintilião"),
+    (1e72, "tresvigintilião"),
+    (1e75, "quattuorvigintilião"),
+    (1e78, "quinquavigintilião"),
+    (1e81, "qesvigintilião"),
+    (1e84, "septemvigintilião"),
+    (1e87, "octovigintilião"),
+    (1e90, "novemvigintilião"),
+    (1e93, "trigintilião"),
+    (1e96, "untrigintilião"),
+    (1e99, "duotrigintilião"),
+    (1e102, "trestrigintilião"),
+    (1e105, "quattuortrigintilião"),
+    (1e108, "quinquatrigintilião"),
+    (1e111, "sestrigintilião"),
+    (1e114, "septentrigintilião"),
+    (1e117, "octotrigintilião"),
+    (1e120, "noventrigintilião"),
+    (1e123, "quadragintilião"),
+    (1e153, "quinquagintilião"),
+    (1e183, "sexagintilião"),
+    (1e213, "septuagintilião"),
+    (1e243, "octogintilião"),
+    (1e273, "nonagintilião"),
+    (1e303, "centilião"),
+    (1e306, "uncentilião"),
+    (1e309, "duocentilião"),
+    (1e312, "trescentilião"),
+    (1e333, "decicentilião"),
+    (1e336, "undecicentilião"),
+    (1e363, "viginticentilião"),
+    (1e366, "unviginticentilião"),
+    (1e393, "trigintacentilião"),
+    (1e423, "quadragintacentilião"),
+    (1e453, "quinquagintacentilião"),
+    (1e483, "sexagintacentilião"),
+    (1e513, "septuagintacentilião"),
+    (1e543, "ctogintacentilião"),
+    (1e573, "nonagintacentilião"),
+    (1e603, "ducentilião"),
+    (1e903, "trecentilião"),
+    (1e1203, "quadringentilião"),
+    (1e1503, "quingentilião"),
+    (1e1803, "sescentilião"),
+    (1e2103, "septingentilião"),
+    (1e2403, "octingentilião"),
+    (1e2703, "nongentilião"),
+    (1e3003, "millinilião")
+])
+
+_ORDINAL_BASE_PT = {
+    1: 'primeiro',
+    2: 'segundo',
+    3: 'terceiro',
+    4: 'quarto',
+    5: 'quinto',
+    6: 'sixto',
+    7: 'sétimo',
+    8: 'oitavo',
+    9: 'nono',
+    10: 'décimo',
+    20: 'vigésimo',
+    30: 'trigésimo',
+    40: "quadragésimo",
+    50: "quinquagesimo",
+    60: "sextagésimo",
+    70: "septagésimo",
+    80: "octagésimo",
+    90: "nonaésimo",
+    10e3: "centésimo",
+    1e3: "milésimo"
+}
+
+_SHORT_ORDINAL_PT = {
+    1e6: "milionésimo",
+    1e9: "bilionésimo",
+    1e12: "trilionésimo",
+    1e15: "quadrilionésimo",
+    1e18: "quintilionésimo",
+    1e21: "sextilionésimo",
+    1e24: "septilionésimo",
+    1e27: "octilionésimo",
+    1e30: "nonilionésimo",
+    1e33: "decilionésimo"
+    # TODO > 1e-33
+}
+_SHORT_ORDINAL_PT.update(_ORDINAL_BASE_PT)
+
+_LONG_ORDINAL_PT = {
+    1e6: "milionésimo",
+    1e12: "bilionésimo",
+    1e18: "trilionésimo",
+    1e24: "quadrilionésimo",
+    1e30: "quintilionésimo",
+    1e36: "sextilionésimo",
+    1e42: "septilionésimo",
+    1e48: "octilionésimo",
+    1e54: "nonilionésimo",
+    1e60: "decilionésimo"
+    # TODO > 1e60
+}
```

Comment:
Shouldn't be here also the femenine forms? It should be modified just like the numbers dictionaries
```
"primeiro": 1,
"primeira": 1,
...
```

Reply:
that's a good point, but i think that would make more sense as a replacement while pronouncing, maybe an helper function to switch_gender

that could also be added into the text normalization step for all numbers so they are extracted correctly

seems strange to maintain versions for plural/singular/female/male if it can be done with a rule based approach